src/libcore/mem/mod.rs

```diff
@@ -68,9 +70,40 @@ pub use crate::intrinsics::transmute;
 /// mem::forget(file);
 /// ```
 ///
-/// The practical use cases for `forget` are rather specialized and mainly come
-/// up in unsafe or FFI code. However, [`ManuallyDrop`] is usually preferred
-/// for such cases, e.g.:
+/// This is useful when the ownership of the underlying was previously
+/// transferred to code outside of Rust, for example by transmitting the raw
+/// file descriptor to C code.
+///
+/// # Relationship with `ManuallyDrop`
+///
+/// Using `mem::forget` to transmit memory ownership is error-prone and is best
+/// replaced with `ManuallyDrop`. Consider, for example, this code:
```

Comment:
```suggestion
/// While `mem::forget` can also be used to transfer *memory* ownership, doing so is error-prone.
/// `ManuallyDrop` should be used instead. Consider, for example, this code:
```
Please also make `ManuallyDrop` here a link to the respective docs.

Reply:
Done.